lib/bookmark_manager.rb

```diff
@@ -7,22 +7,27 @@ def initialize(user)
     @user = user
   end
 
-  def create(post_id:, name: nil, reminder_type: nil, reminder_at: nil, options: {})
-    post = Post.find_by(id: post_id)
+  def create(post_id:, topic_id: nil, name: nil, reminder_type: nil, reminder_at: nil, options: {})
```

Comment:
```suggestion
  def create(topic:,  post: nil, name: nil, reminder_type: nil, reminder_at: nil, options: {})
```

I may even go so far to suggest that the controller handles the loading and pass the active record objects through to this method.